src/gfn/actions.py

```diff
@@ -99,37 +91,37 @@ def extend_with_dummy_actions(self, required_first_dim: int) -> None:
             n = required_first_dim - self.batch_shape[0]
             dummy_actions = self.__class__.make_dummy_actions((n, self.batch_shape[1]))
             self.batch_shape = (self.batch_shape[0] + n, self.batch_shape[1])
-            self.actions_tensor = torch.cat(
-                (self.actions_tensor, dummy_actions.actions_tensor), dim=0
-            )
+            self.tensor = torch.cat((self.tensor, dummy_actions.tensor), dim=0)
         else:
             raise NotImplementedError(
                 "extend_with_dummy_actions is only implemented for bi-dimensional actions."
             )
 
-    def compare(self, other: ActionsTensor) -> BoolTensor:
+    def compare(
+        self, other: TT["batch_shape", "action_shape"]
+    ) -> TT["batch_shape", torch.bool]:
         """Compares the actions to a tensor of actions.
         Args:
             other: tensor of actions
         Returns:
             boolean tensor of shape batch_shape indicating whether the actions are equal
         """
-        out = self.actions_tensor == other
+        out = self.tensor == other.tensor
         action_ndim = len(self.__class__.action_shape)
-        for _ in range(action_ndim):
+        for _ in range(action_ndim):  # TODO: this seems over complex?
```

Comment:
do you know of a simpler alternative ? It felt weird writing these lines indeed

Reply:
looking at it now!

Reply:
Check this out - we can flatten all actions dims assuming the batch dims are always defined: https://github.com/saleml/torchgfn/pull/47/commits/3a85dd9cebdc80a2160ee8125484f438204655f6

Reply:
Awesome !